test/parallel/test-stdin-script-child-option.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const expected = '--option-to-be-seen-on-child';
+
+const spawn = require('child_process').spawn;
+const child = spawn(process.execPath, ['-', expected], { stdio: 'pipe' });
+
+child.stdin.end('console.log(process.argv[2])');
+
+let actual = '';
+child.stdout.on('data', (chunk) => actual += chunk);
```

Comment:
You should call `child.stdout.setEncoding('utf8');` before using `.on('data')` with strings